src/main/java/com/tranhiep/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.tranhiep.service.impl;
+
+import com.tranhiep.dao.BookDAO;
+import com.tranhiep.dao.impl.BookDAOImpl;
+import com.tranhiep.entity.BookEntity;
+import com.tranhiep.service.BookService;
+
+import javax.persistence.EntityManager;
+import java.util.Collection;
+
+
+public class BookServiceImpl implements BookService {
+    private final EntityManager entityManager;
+    private final BookDAO bookDAO;
+
+    public BookServiceImpl(EntityManager entityManager){
+        this.entityManager = entityManager;
+        bookDAO = new BookDAOImpl(entityManager);
+    }
+
+
+    @Override
+    public Collection<BookEntity> getAll() {
+        Collection<BookEntity> bookEntities =  bookDAO.getAll(BookEntity.class);
+
+        //get author for each book
+        bookEntities.forEach(e -> e.getAuthors());
```

Comment:
`bookEntities.forEach(e -> e.getAuthors());`
Please explain why need this statement?

Reply:
dạ anh. tại ở entity em để fetch lazy nên khi lần đầu bookDAO.getAll() sẽ không có lấy author trong mỗi book nên gọi e.getAuthors() để lấy được các author của mỗi book.

Reply:
Mỗi lần em danh sách book, thì em lại loop để lấy author như vậy thì trong trường hợp này để lazy fetch có lợi gì? Tại sao không để eager ngay từ đầu luôn đi.

Reply:
dạ vâng a. tại lúc đầu em tính show list book mà không cần authors, mà lúc sau lại cần nên e gọi vậy.